File: DareFightingICE/SampleAI/PythonSamples/KickAI.py
from pyftg.ai_interface import AIInterface
from pyftg.struct import *

class KickAI(AIInterface):
    def __init__(self):
        super().__init__()
        self.blind_flag = True

    def name(self) -> str:
        return self.__class__.__name__

    def is_blind(self) -> bool:
        return self.blind_flag

    def initialize(self, game_data: GameData, player_number: int):
        self.cc = CommandCenter()
        self.key = Key()
        self.player = player_number

    def input(self) -> Key:
        return self.key

    def get_information(self, frame_data: FrameData, is_control: bool, non_delay_frame_data: FrameData):
        self.frame_data = frame_data
        self.cc.set_frame_data(self.frame_data, self.player)

    def get_screen_data(self, screen_data: ScreenData):
        self.screen_data = screen_data

    def get_audio_data(self, audio_data: AudioData):
        self.audio_data = audio_data

    def processing(self):
        if self.frame_data.empty_flag or self.frame_data.current_frame_number <= 0:
            return

        if self.cc.get_skill_flag():
            self.key = self.cc.get_skill_key()
        else:
            self.key.empty()
            self.cc.skill_cancel()

            self.cc.command_call("B")
    
    def round_end(self, round_result: RoundResult):
        print(round_result.remaining_hps[0])
        print(round_result.remaining_hps[1])
        print(round_result.elapsed_frame)

    def game_end(self):
        pass

File: DareFightingICE/SampleAI/JavaSamples/src/KickAI.java
import aiinterface.AIInterface;
import aiinterface.CommandCenter;
import struct.AudioData;
import struct.FrameData;
import struct.GameData;
import struct.Key;
import struct.RoundResult;
import struct.ScreenData;

public class KickAI implements AIInterface {

	private boolean blindFlag;
	private boolean playerNumber;
	private FrameData frameData;
	private Key key;
	private CommandCenter cc;
	
	public KickAI() {
		this.blindFlag = false;
	}
	
	@Override
	public String name() {
		return this.getClass().getName();
	}
	
	@Override
	public boolean isBlind() {
		return this.blindFlag;
	}

	@Override
	public void initialize(GameData gameData, boolean playerNumber) {
		this.playerNumber = playerNumber;
		
		this.key = new Key();
		this.cc = new CommandCenter();
	}
	
	@Override
	public void getInformation(FrameData frameData, boolean isControl, FrameData nonDelay) {
		this.frameData = frameData;
		this.cc.setFrameData(frameData, playerNumber);
	}
	
	@Override
	public void getScreenData(ScreenData screenData) {
		
	}

	@Override
	public void getAudioData(AudioData audioData) {
		
	}

	@Override
	public Key input() {
		return this.key;
	}

	@Override
	public void processing() {
		if (frameData.getEmptyFlag() || frameData.getFramesNumber() <= 0) {
			return;
		}
		
		if (cc.getSkillFlag()) {
			key = cc.getSkillKey();
		} else {
			key.empty();
			cc.skillCancel();
			
			cc.commandCall("B");
		}
	}
	
	@Override
	public void roundEnd(RoundResult roundResult) {
		System.out.println(roundResult.getRemainingHPs()[0] + " " + roundResult.getRemainingHPs()[1] + " " + roundResult.getElapsedFrame());
	}
	
	@Override
	public void gameEnd() {
		System.out.println("game end");
	}
	
}
